Realm/Realm.Weaver/Extensions/TypeReferenceExtensions.cs

```diff
@@ -17,12 +17,35 @@
 ////////////////////////////////////////////////////////////////////////////
 
 using System.ComponentModel;
+using System.Linq;
 using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
 using RealmWeaver;
 
 [EditorBrowsable(EditorBrowsableState.Never)]
 internal static class TypeReferenceExtensions
 {
+    public static SequencePoint GetSequencePoint(this TypeDefinition @this)
+    {
+        return GetCtorSequencePoint() ?? GetPropSequencePoint();
+
+        SequencePoint GetCtorSequencePoint()
+        {
+            return @this.GetConstructors()
+                .OrderBy(c => c.Parameters.Count)
+                .Select(c => c.DebugInformation.SequencePoints.FirstOrDefault())
+                .FirstOrDefault(sp => sp != null);
```

Comment:
What's the difference between just calling this `FirstOrDefault` instead of having it twice?

Reply:
Not sure I understand. If you're talking about the `FirstOrDefault` on line 37, then those are different. On line 37, `FirstOrDefault()` gives us the first element from `DebugInformation.SequencePoints` and puts that into our collection. The `FirstOrDefault` on line 38  gives us the first element from our collection. We could rewrite that as:

```
@this.GetConstructors()
    .OrderBy(c => c.Parameters.Count)
    .SelectMany(c => c.DebugInformation.SequencePoints)
    .FirstOrDefault()
```

Reply:
Yes, that's what I meant. With the other example I do understand the difference now. I guess I need to read up on LINQ a bit more so that I get used to it. 👍